shadowreader/plugins/elb.py

```diff
@@ -0,0 +1,208 @@
+"""
+Copyright 2018 Edmunds.com, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+"""
+
+import gzip
+import re
+import boto3
+
+from libs.s3 import (
+    put_on_s3,
+    _generate_key_prefix_base_on_time,
+    delete_tmp_file,
+    fetch_file_names_on_s3,
+)
+from classes.mytime import MyTime
+from datetime import datetime
+from typing import Tuple, Match
+
+s3cl = boto3.client("s3")
+elb_regex = (
+    '(?P<timestamp>[\S]+) (?P<elb>[\S]+) (?P<client>[\S]+):(?P<client_port>[\S]+) (?P<target>[\S]+):(?P<target_port>[\S]+) (?P<request_processing_time>[\S]+) (?P<target_processing_time>[\S]+) (?P<response_processing_time>[\S]+) (?P<elb_status_code>[\S]+) (?P<target_status_code>[\S]+) (?P<received_bytes>[\S]+) (?P<sent_bytes>[\S]+) (?P<request>".*") (?P<user_agent>".*") (?P<ssl_cipher>[\S]+) (?P<ssl_protocol>[\S]+)'
+)
+regex = re.compile(elb_regex)
+
+
+def _download_file(bucket: str, key: str) -> str:
+    """ download file, unzip it then return the contents """
+    tmp_file_name = f"/tmp/logs"
+
+    try:
+        with open(tmp_file_name, "wb") as data:
+            s3cl.download_fileobj(bucket, key, data)
+    except Exception as e:
+        print(type(e).__name__, e)
+        f = open(tmp_file_name, "w")
```

Comment:
I noticed sometimes the call to S3 will throw an exception and download a corrupted file, so I wanted to clear such files before it being opened in the next lines. But it may be cleaner to just delete it